src/NServiceBus.Transport.Msmq/MessagePump.cs

```diff
@@ -159,11 +159,12 @@ public void Start(PushRuntimeSettings limitations)
                     // antecedent task is aborted the continuation will be scheduled. But in this case we don't need to await
                     // the continuation to complete because only really care about the receive operations. The final operation
                     // when shutting down is a clear of the running tasks anyway.
-                    receiveTask.ContinueWith((t, state) =>
+                    void RemoveReceiveTask(Task task, object state)
```

Comment:
Do we even need the state object when we use a local function? The state object was added to avoid the closure.

Reply:
Good catch. The local functions becomes a method (I think) and no closure needs to be created.

The fix is on the way.